pandas/core/index.py

```diff
@@ -2226,7 +2228,9 @@ def drop(self, labels):
         indexer = self.get_indexer(labels)
         mask = indexer == -1
         if mask.any():
-            raise ValueError('labels %s not contained in axis' % labels[mask])
```

Comment:
This is kind of a gray area to me. We're not doing actual indexing here. I would probably stick with `ValueError`, especially to remain consistent with the existing implementation.